pkg/controller.v1/mxnet/mxjob_controller.go

```diff
@@ -178,46 +186,46 @@ func (r *MXJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *MXJobReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	// setup FieldIndexer to inform the manager that this controller owns pods and services,
-	// so that it will automatically call Reconcile on the underlying MXJob when a Pod or Service changes, is deleted, etc.
-	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &corev1.Pod{}, jobOwnerKey, func(rawObj client.Object) []string {
-		pod := rawObj.(*corev1.Pod)
-		owner := metav1.GetControllerOf(pod)
-		if owner == nil {
-			return nil
-		}
+	c, err := controller.New(r.ControllerName(), mgr, controller.Options{
+		Reconciler: r,
+	})
 
-		// Make sure owner is XGBoostJob Controller.
-		if owner.APIVersion != r.GetAPIGroupVersion().Version || owner.Kind != r.GetAPIGroupVersionKind().Kind {
-			return nil
-		}
-
-		return []string{owner.Name}
-	}); err != nil {
+	if err != nil {
 		return err
 	}
 
-	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &corev1.Service{}, jobOwnerKey, func(rawObj client.Object) []string {
-		svc := rawObj.(*corev1.Service)
-		owner := metav1.GetControllerOf(svc)
-		if owner == nil {
-			return nil
-		}
+	// using onOwnerCreateFunc is easier to set defaults
+	if err = c.Watch(&source.Kind{Type: &mxjobv1.MXJob{}}, &handler.EnqueueRequestForObject{},
```

Comment:
Any reason we prefer using `Watch` instead of `Owns`?

Reply:
`Watch` is the low-level functions provided by builder. `Owns` and `For` can be replaced by `Watch` with some enqueue handlers. 

We want to leverage `onOwnerCreateFunc` to set defaults instead of using Kubebuilder way (webhook).  I notice owns can not provide pluggable handler.  

If there's a way to do that, let me know

Reply:
I see. It makes sense.